### ISSUE TITLE
Error 3070003: Serialization Error Processing WASM

### ISSUE THREAD

**Issue by Lynette-xin** [EOSIO/eosio.cdt#321]
compile cpp: eosio-cpp -abigen hello.cpp -o hello.wasm
set contract: cleos set contract hello ./hello -p hello@active
I get : Error 3070003: Serialization Error Processing WASM

**Comment by Lynette-xin** [comment]
I think may be this error is because the update of eosio-cpp which previous version is eosiocpp,then the new version eosio-cpp compile the file of hello.cpp ,and create a result fie hello.wasm can't be acceptable by current version of nodeos,so this error occur

**Comment by pietromartinso** [comment]
I've run nodeos and keosd within docker containers. The eosio versions seems to be v1.5.0-rc1 (github tag 5985cc40 - got with "cleos version client").

First, when I try to compile my contract, with command: "eosio-cpp -o hello.wasm hello.cpp --abigen", compilation process kind of "get stuck" and I'm not able to take back control of my prompt without using ctrl+c. Unfortunately, when I stop the compilation process with "ctrl+c", it doesn't produce the abi file, and produced binary (wasm) seems to be corrupted.

Alternatively, when I compile with -c option (command "eosio-cpp -o hello.wasm hello.cpp --abigen -c"), it seems to work fine and the abi plus wasm files appearently are generated correctly. 

After that, when I try to deploy the contract, with a command like "cleos --url http://jungle2.cryptolions.io:80 --wallet-url http://172.18.0.3:9876 set contract pietroconta5 /mnt/dev/data/hello -p pietroconta5", I face the same error, again and again:

Reading WASM from /mnt/dev/data/hello/hello.wasm...
Publishing contract...
Error 3070003: Serialization Error Processing WASM
Error Details:
mutable globals cannot be imported: globalImport.type.isMutable
pending console output: 


-----------------------
I'm pretty sure I have my accounts keys imported to my wallet and my wallet is open and unlocked, with the accounts respective keys. Also, I ensure that my accounts holds some enough EOS tokens.

I have tried to deploy my contract locally, on my block producer and also tried to deploy at jungle testnet (endpoint: http://jungle2.cryptolions.io:80, and also tested others).

//-----------------------------
//My hello.hpp file (contract header) is:
#include <eosiolib/eosio.hpp>

CONTRACT hello : public eosio::contract {
public:
  using contract::contract;

  ACTION hi(eosio::name user);
};

EOSIO_DISPATCH( hello, (hi))
//-----------------------------
//My hello.cpp file (contract implementation) is:
#include <eosiolib/system.h>
#include “hello.hpp”

void hello::hi(eosio::name user) {
  eosio::print(“eae vacilao: ”, eosio::name(user));
}
-----------------------------------
The ABI generated file is:
----------------------------------
{
    "____comment": "This file was generated with eosio-abigen. DO NOT EDIT Sun Dec 16 16:01:59 2018",
    "version": "eosio::abi/1.1",
    "structs": [
        {
            "name": "hi",
            "base": "",
            "fields": [
                {
                    "name": "user",
                    "type": "name"
                }
            ]
        }
    ],
    "types": [],
    "actions": [
        {
            "name": "hi",
            "type": "hi",
            "ricardian_contract": ""
        }
    ],
    "tables": [],
    "ricardian_clauses": [],
    "variants": [],
    "abi_extensions": []
}

**Comment by wumao** [comment]
mee too, how to fix this. 
root@vultr:~/eos/contracts/hello# cleos set contract hello /root/eos/contracts/hello -p hello@active
Reading WASM from /root/eos/contracts/hello/hello.wasm...
Publishing contract...
Error 3070003: Serialization Error Processing WASM
Error Details:
mutable globals cannot be imported: globalImport.type.isMutable
pending console output:

**Comment by wenzhenxiang** [comment]
I have the same issue on a VPS with only 1 CPU, after I switching to a VPS with 2 CPU. 
All is ok. just need switching 2CPU

**Comment by agxmeister** [comment]
It does not work even on 2 CPUs machine:

cleos set contract hello . hello.wasm hello.abi -p hello@active
Reading WASM from /home/aleksey/git/contracts/hello/hello.wasm...
Publishing contract...
Error 3070003: Serialization Error Processing WASM
Error Details:
invalid index: localIndex>=locals.size()

**Comment by Tommo-L** [comment]
The same error in my env:  centos7， memory 8G,  1CPU 6Cores

```
Reading WASM from /root/workspace/contracts/hello/hello.wasm...
Publishing contract...
Error 3070003: Serialization Error Processing WASM
Error Details:
invalid index: localIndex>=locals.size()
pending console output:
```


Dose anyone solve this problem?

**Comment by agxmeister** [comment]
For me this issue was solved by updating CDT to version 1.6.1.

**Comment by Tommo-L** [comment]
Tks， I'll try it.

**Comment by jchung00** [comment]
@Tommo-L did this solve the problem?

**Comment by eewindfly** [comment]
it fix my problem by updating CDT to 1.6.2

**Comment by seanow** [comment]
same here. Fix by updating to 1.6.2. It's irrelevant to CPU count in my case.

**Comment by jeffreyssmith2nd** [comment]
Closing this since it sounds like 1.6.2 has resolved the issue.
